crates/stackable-operator/src/logging/mod.rs

```diff
@@ -22,18 +25,32 @@ impl Default for TracingTarget {
 /// We force users to provide a variable name so it can be different per product.
 /// We encourage it to be the product name plus `_LOG`, e.g. `FOOBAR_OPERATOR_LOG`.
 /// If no environment variable is provided, the maximum log level is set to INFO.
+///
+/// Log output can be copied to a file by setting `{env}_DIRECTORY` (e.g. `FOOBAR_OPERATOR_DIRECTORY`)
+/// to a directory path. This file will be rotated regularly.
 pub fn initialize_logging(env: &str, app_name: &str, tracing_target: TracingTarget) {
     let filter = match EnvFilter::try_from_env(env) {
         Ok(env_filter) => env_filter,
         _ => EnvFilter::try_new(tracing::Level::INFO.to_string())
             .expect("Failed to initialize default tracing level to INFO"),
     };
 
-    let fmt = tracing_subscriber::fmt::layer();
-    let registry = Registry::default().with(filter).with(fmt);
+    let terminal_fmt = tracing_subscriber::fmt::layer();
+
+    let file_appender_directory = std::env::var_os(format!("{env}_DIRECTORY")).map(PathBuf::from);
+    let file_fmt = file_appender_directory.as_deref().map(|log_dir| {
+        let file_appender = RollingFileAppender::builder()
+            .rotation(Rotation::HOURLY)
+            .filename_suffix(format!("{app_name}.tracing-rs.json"))
```

Comment:
`app_name` should probably be the filename prefix:

```suggestion
            .filename_prefix(app_name.to_string())
            .filename_suffix("tracing-rs.json")
```

This generates filenames like `opa-bundle-builder.2024-07-25-07.tracing-rs.json`.

Reply:
I agree on a level, but this would be a behaviour change that I'm not sure how Vector reacts to. Previously, Vector would only have the "current" file listed for ingestion. With this change, it would also see all already-rotated files.

Reply:
My suggested change is actually only a cosmetic one. With the change, the filename is `opa-bundle-builder.2024-07-25-07.tracing-rs.json` and without it, it is `2024-07-25-07.opa-bundle-builder.tracing-rs.json`. This makes no difference in processing.

The current implementation creates a new file every hour. Vector starts watching it and still watches the old ones, which is no problem:

```
Found new file to watch. file=/stackable/log/bundle-builder/2024-07-25-11-52.opa-bundle-builder.tracing-rs.json
Found new file to watch. file=/stackable/log/bundle-builder/2024-07-25-11-53.opa-bundle-builder.tracing-rs.json
Found new file to watch. file=/stackable/log/bundle-builder/2024-07-25-11-54.opa-bundle-builder.tracing-rs.json
Found new file to watch. file=/stackable/log/bundle-builder/2024-07-25-11-55.opa-bundle-builder.tracing-rs.json
Found new file to watch. file=/stackable/log/bundle-builder/2024-07-25-11-56.opa-bundle-builder.tracing-rs.json
Found new file to watch. file=/stackable/log/bundle-builder/2024-07-25-11-57.opa-bundle-builder.tracing-rs.json

```

When more than 6 log files exist, the oldest one is deleted, and Vector stops watching it:

```
Found new file to watch. file=/stackable/log/bundle-builder/2024-07-25-11-58.opa-bundle-builder.tracing-rs.json
Stopped watching file. file=/stackable/log/bundle-builder/2024-07-25-11-52.opa-bundle-builder.tracing-rs.json reached_eof="true"
```

This is even simpler than with the other logging frameworks which rename the log files on roll-over.

Reply:
Oh right, I mixed up `prefix` and `suffix` in the original implementation >.<

But sure, if it works I'm happy to change the format.